### PR TITLE
Organised the code and chose meaningful names for variables. We restr…

### PR DESCRIPTION
**Pull request by fkazemian**
…uctured the main loop and separated it into several functions.

### CODE REVIEW

**Comment by jstac** [comment]
@fkazemian Many thanks!  This is **much** better.

Could you please read [this style guide](https://docs.julialang.org/en/stable/manual/style-guide/) carefully.  It will suggest some further changes.  For example, the function `mainloop` modifies its arguments and hence should be `mainloop!`.  In fact I suggest you call it `bellman_operator!`.  

Once you are done please let me know and I will merge this PR.

**Comment by scstearn** [comment]
Only a couple of things. Firstly, I understand that using Greek alphabet characters is shorter and more readable, but it makes editing code a lot slower if a special character palette is required. But perhaps I'm being a little naive- do you have a way of entering such characters quickly from the keyboard alone?

Typically performance loss is minimal for successive function calls and thus negligible, but it's worth bearing in mind. I will see if it has any effect in this case- it probably won't. Depends how many times the mainloop function is called on average. There are bigger things to worry about though.

Otherwise looks good!

**Comment by scstearn** [comment]
That complies more with the Julia style guide, good work.

**Comment by jstac** [comment]
@scstearn Greek letters can be entered in Julia-aware editors using `\alpha` and hitting the tab key.  I use vim with the julia-vim plug in and it works nicely:

https://github.com/JuliaEditorSupport/julia-vim#latex-to-unicode-substitutions

It also works well in Jupyter.  I'm in favor of using unicode symbols, despite the hassle, because they help bring the problem closer to the maths.

**Comment by scstearn** [comment]
That's cool- all for it. Thanks for the information. Currently using the uber-juno IDE plugin for Atom. Seems to work there too.

**Comment by jstac** [comment]
@fkazemian Thanks, nice work.  I'll merge this PR now.

BTW I often organize my Julia code by keeping the important functions and type declarations in a *.jl file and then the code that runs the functions in Jupyter notebooks.  I end up with a collection of notebooks that run the functions to produce various figures, output, timing of code, etc.  See [here](https://github.com/jstac/recursive_utility_code) for example.

It's up to you, but you might find that structure useful.

**Comment by jstac** [comment]
@fkazemian @scstearn I propose you adopt a system whereby, if one person submits a PR, the other person merges it when they think it's ready.  You don't need me to be a part of that work flow, although I'll keep track of the PRs and add comments where I think I can help.

**Comment by scstearn** [comment]
Sounds good. We can do that.

File: Basic_RBC6.jl [hunk 1]
```diff
@@ -1,106 +1,141 @@
...
+        E = zeros(n,m)
+        output = (k_grid.^α)*z;
+        #-----------------------------------------------------------------
+        error = 10.0
+        tol = 0.0000001
+        iter = 0
+        #-----------------------------------------------------------------
+        while error > tol
+            E = f_value*T';
+            bellman_operator!(p ,Matrix(output),Matrix(f_value),Matrix(f_valueN),Matrix(f_policy),Matrix(E))
```
Comment (sglyon) [inline Basic_RBC6.jl:127]: Why the call to `Matrix` here?

I believe that given how you instantiated these variables above they should always be matrices. If that isn't true then there are probably other issues we should look into.

Comment (fkazemian) [reply Basic_RBC6.jl:127]: Thank you, I followed the instruction from "Handle excess argument diversity in the caller" in this tutorial https://docs.julialang.org/en/stable/manual/style-guide/

**Comment by fkazemian** [comment]
@jstac  Thank you. That would be great.

**Pull request merged**